src/api/trace/span.rs

```diff
@@ -38,6 +39,42 @@ pub trait Span: fmt::Debug + 'static + Send + Sync {
         self.add_event_with_timestamp(name, SystemTime::now(), attributes)
     }
 
+    /// Convenience method to record an exception/error as an `Event`
+    ///
+    /// An exception SHOULD be recorded as an Event on the span during which it occurred.
+    /// The name of the event MUST be "exception".
+    ///
+    /// The semantic conventions for Errors are described in ["Semantic Conventions for Exceptions"](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/exceptions.md)
+    ///
+    /// If user doesn't provide `error_type` or `stacktrace`, the corresponding attributes will not be set.
+    ///
+    /// User can custom exception message by overriding error's `fmt` method.
+    fn record_exception(
+        &self,
+        err: &dyn Error,
+        error_type: Option<String>,
+        stacktrace: Option<String>,
+    ) {
+        // build up attributes
+        let mut attributes = vec![api::KeyValue::new(
+            "exception.message",
+            format!("{:?}", err),
+        )];
+
+        // if user provide stacktrace, then we set it.
+        // In the future, we can use `Error::stacktrace()` method if it's stabilized.
```

Comment:
Probably want `Error::backtrace` in this comment.

Reply:
Sorry kind of a typo.